addons/fix-costume-drag/userscript.js

```diff
@@ -0,0 +1,181 @@
+      if (disabled) {
+        return;
+      }
+
+      const scrollContainer = this.ref.querySelector('[class*="selector_list-area"]');
+
+      if (newProps.dragInfo.dragging && !this.props.dragInfo.dragging) {
+        // When started dragging
+        this.initialScrollTop = scrollContainer.scrollTop;
+
+        // Add scroll listener
+        const onScroll = () => {
+          if (this.props.dragInfo.dragging) {
+            this.forceUpdate();
+          }
+        };
+
+        scrollContainer.addEventListener("scroll", onScroll);
+
+        // Save listener so it can be removed later
+        this._scrollListener = onScroll;
+      } else if (!newProps.dragInfo.dragging && this.props.dragInfo.dragging) {
+        // When stopped dragging
+        if (this._scrollListener) {
+          scrollContainer.removeEventListener("scroll", this._scrollListener);
+          this._scrollListener = null;
+        }
+      }
+    };
+
+    // While dragging
+    SortableHOC.prototype.getMouseOverIndex = function () {
+      // Just call original function if disabled
+      if (disabled) {
+        return originalGetMouseOverIndex.call(this);
+      }
+
+      let index = null;
+      if (this.props.dragInfo.currentOffset) {
+        const scrollContainer = this.ref.querySelector('[class*="selector_list-area"]');
+        const x = this.props.dragInfo.currentOffset.x; // x isn't affected
+        const y =
+          this.props.dragInfo.currentOffset.y +
+          scrollContainer.scrollTop -
+          scrollContainer.getBoundingClientRect().top -
+          this.initialScrollTop +
+          96;
+
+        if (this.boxes.length === 0) {
+          index = 0;
+        } else {
+          index = indexForPositionOnList({ x, y }, this.boxes, this.props.isRtl);
+        }
+
+        // Setting Drag at top/bottom to scroll
+        if (dragScrollSetting) {
+          const containerRect = scrollContainer.getBoundingClientRect();
+          const edgeSize = 30; // Distance from the top/bottom to trigger scroll
+
+          if (this.props.dragInfo.currentOffset.y < containerRect.top + edgeSize) {
+            scrollContainer.scrollTop -= scrollSpeed;
+          } else if (this.props.dragInfo.currentOffset.y > containerRect.bottom - edgeSize) {
+            scrollContainer.scrollTop += scrollSpeed;
+          }
+        }
+      }
+      return index;
+    };
+  };
+
+  // When changed settings
+  addon.settings.addEventListener("change", function () {
+    dragScrollSetting = addon.settings.get("drag-scroll");
+    scrollSpeedSetting = addon.settings.get("scroll-speed");
+    switch (scrollSpeedSetting) {
+      case "slow":
+        scrollSpeed = SLOW_SCROLL_SPEED;
+        break;
+      case "fast":
+        scrollSpeed = FAST_SCROLL_SPEED;
+        break;
+      default:
+        scrollSpeed = DEFAULT_SCROLL_SPEED;
+    }
+  });
+
+  // When enabled/disabled function
+  addon.self.addEventListener("disabled", function () {
+    disabled = true;
```

Comment:
Checking `addon.self.disabled` directly is very cheap so it's fine to use it every time.